compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEGenerateControlOverlay.cpp

```diff
@@ -46,9 +46,66 @@ LogicalResult initializeChannelsGenerators(
   return success();
 }
 
+/// Establishes a packet-mode connection between the source tile DMA and the
+/// target tile CTRL ports. If multiple target tiles are provided, the
+/// connection is configured as a broadcast.
+LogicalResult buildShimTileToCtrlConnection(
+    IRRewriter &rewriter, AMDAIE::ControlCodeOp controlCodeOp,
+    AMDAIE::TileOp srcTileOp, ArrayRef<TileOp> targetTileOps,
+    DenseMap<Value, ChannelGenerator> &shimTileToGeneratorMap) {
+  // Get the available DMA channel for the shim tile, and assign it for the
+  // packet flow.
+  std::optional<uint8_t> maybeChannel =
+      shimTileToGeneratorMap[srcTileOp.getResult()]
```

Comment:
I think it would be good to check here that `srcTileOp.getResult()` exists in `shimTileToGeneratorMap` and/or that `srcTileOp` is a Shim tile and return an error otherwise.